pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -150,12 +162,58 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		reasonsForWorkerPoolKubernetesUpdate[w.Name] = reasonForWorkerPoolKubernetesUpdate
 	}
 
-	maintainOperation(shoot)
+	for workerPool, reason := range reasonsForWorkerPoolKubernetesUpdate {
+		if reason != "" {
+			operation := fmt.Sprintf("For worker pool %s: %s", workerPool, reason)
+			operations = append(operations, operation)
+		}
+	}
+
+	operation := maintainOperation(maintainedShoot)
+	if operation != "" {
+		operations = append(operations, "Other task started during maintenance time - "+operation)
+	}
+
+	if operations != nil {
```

Comment:
This is more intuitive here, I think:

```suggestion
	if len(operations) > 0 {
```